src/transactions/OfferTests.cpp

```diff
@@ -1096,38 +1235,38 @@ TEST_CASE("create offer", "[tx][offers]")
                 });
 
                 for_versions_from(10, *app, [&]() {
-                    // as b1 cannot buy any more IDRs the offer should be
-                    // deleted
-                    auto offerB1 = market.requireChangesWithOffer(
-                        {{offerC1.key, offerC1Changed}}, [&] {
-                            return market.addOffer(b1, offerB1Params,
-                                                   OfferState::DELETED);
-                        });
+                    // as b1 cannot buy any more IDRs the operation should fail
+                    REQUIRE_THROWS_AS(
+                        market.addOffer(b1, offerB1Params, OfferState::DELETED),
+                        ex_MANAGE_OFFER_LINE_FULL);
                 });
             }
             SECTION("Offer reaches limit")
             {
-                // make it that c1 can only receive 9 USD (for 1 IDR)
-                c1.changeTrust(usd, 9 * assetMultiplier);
-                // make it that b1 can receive more than 1 IDR
-                b1.changeTrust(idr, 1000 * assetMultiplier);
-
-                // as c1 cannot buy any more USDs, c1's offer should be
-                // deleted
-                // b1's offer is created to sell the remainder
-                // 200-9 = 191 USD
-
-                auto offerRemaining =
-                    OfferState{usd, idr, p, 191 * assetMultiplier};
-
-                for_all_versions(*app, [&]() {
+                for_versions_to(9, *app, [&]() {
+                    // make it that c1 can only receive 9 USD (for 1 IDR)
+                    c1.changeTrust(usd, 9 * assetMultiplier);
+                    // make it that b1 can receive more than 1 IDR
+                    b1.changeTrust(idr, 1000 * assetMultiplier);
 
+                    // as c1 cannot buy any more USDs, c1's offer should be
+                    // deleted
+                    // b1's offer is created to sell the remainder
+                    // 200-9 = 191 USD
+                    auto offerRemaining =
+                        OfferState{usd, idr, p, 191 * assetMultiplier};
                     auto offerB1 = market.requireChangesWithOffer(
                         {{offerC1.key, OfferState::DELETED}}, [&] {
                             return market.addOffer(b1, offerB1Params,
                                                    offerRemaining);
                         });
                 });
+
+                for_versions_from(10, *app, [&]() {
+                    // make it that c1 can only receive 9 USD (for 1 IDR)
+                    REQUIRE_THROWS_AS(c1.changeTrust(usd, 9 * assetMultiplier),
```

Comment:
same thing here: the way the setup is done is now impossible, probably clearer to not test this anymore like that and instead rely on `changeTrust` tests

Reply:
Agreed.